### PR TITLE
Adds instructions for configuring custom tools in Octopus Cloud

### PR DESCRIPTION
**Pull request by geofflamrock**
As part of recent efforts to actively discourage customers using built-in tools some changes were made to add warnings to the UI and deployment logs for some Azure steps, and to instead provision their own tools on workers. This causes some issues for dynamic workers in Octopus Cloud as installing custom tools is not possible.

This PR adds instructions to the Running Azure Powershell docs indicating how to configure custom Azure tools when using dynamic worker pools. It also removes the OctopusTreatWarningsAsErrors system variable as this is no longer used.

### CODE REVIEW

**Review by pawelpabich — COMMENTED** [review]
LGTM + comment

File: docs/deployments/azure/running-azure-powershell/index.md [hunk 1]
```diff
@@ -36,6 +38,13 @@ Azure supports two authentication methods, each of which provides access to a di
 
 Learn more about [configuring the right Azure Account](/docs/infrastructure/deployment-targets/azure/index.md).
 
+## Dynamic Worker Pools
+
+Octopus Cloud uses a special type of worker pool called a [Dynamic Worker Pool](/docs/infrastructure/workers/dynamic-worker-pools.md). These are provided by Octopus and custom versions of the Azure tools cannot be installed on them. In order to use your own version of the Azure CLI or Azure Powershell cmdlets when using Dynamic Worker Pools, please do the following:
```
Comment (pawelpabich) [inline docs/deployments/azure/running-azure-powershell/index.md:43]: `cannot be installed` -> `cannot be easily installed`.  Technically speaking someone could add a piece of code to every step of their process that installs Azure dependencies if they are not installed.

**Comment by geofflamrock** [comment]
[Using Bundled Tools with Azure Steps does not work when using PowerShell Core](https://trello.com/c/DTMxmwiE/29-using-bundled-tools-with-azure-steps-does-not-work-when-using-powershell-core)

**Review by andrewabest — APPROVED** [review]
The changes align with what we discussed - looks good to go!

**Comment by harrisonmeister** [comment]
Late to the party :smile: - FYI - I made minor line edits after this was merged :+1:

**Pull request merged**